pkg/expression/builtin_regexp.go

```diff
@@ -583,8 +587,11 @@ func (re *builtinRegexpSubstrFuncSig) vecEvalString(ctx EvalContext, input *chun
 
 		if !memorized {
 			// Get pattern and match type and then generate regexp
-			pattern := params[1].getStringVal(i)
 			matchType := params[4].getStringVal(i)
+			pattern := params[1].getStringVal(i)
```

Comment:
Maybe we can check empty pattern in `buildRegexp`?

Reply:
Nice suggestion, I'll change it